docs/tools/mbed_cli_2/migration_guide.md

```diff
@@ -0,0 +1,195 @@
+
+* `--scm`: We now always use git.
+* `--program`: There is no need to specify this. We always assume a program.
+* `--library`: There is no need to specify this. We always assume a program.
+* `--mbedlib`: Importing Mbed 2 is no longer supported.
+* `--depth`: This is no longer needed, as Mbed CLI 2 automatically uses the
+  shallowest clone or fetch possible.
+* `--protocol`: This is now determined from the URL instead.
+* `--offline`: This is not needed, as there is no longer any Mbed-specific git
+  cache.
+* `--no-requirements`: Mbed CLI 2 never automatically installs requirements.
+</span><br>
+
+### Importing an Mbed program
+<span style="background-color:#ffe6cc;">The subcommand `mbed-tools
+import` clones an Mbed project and library dependencies. The following
+command-line-options supported in Mbed CLI 1 are deprecated:
+
+* `--depth`: The new CLI opportunistically imports the most shallow repository
+  possible. Running `git clone --depth <depth> <repo_url> | cd <repo_name> &&
+  mbed-tools deploy` will create an Mbed project with the required depth.
+* `--protocol`: To select the protocol, use a URL with the
+  desired protocol. For instance, you may import projects or libraries over SSH
+  (Secure Shell) as follows `mbed-tools import
+  git@github.com:ARMmbed/<some_driver>`, or with https `mbed-tools import
+  https://github.com/ARMmbed/<some_driver>`. Alternatively, if you would prefer
+  to always use SSH, even when your library or its dependencies specify https
+  in their `.lib` files, you can add the following into your .gitconfig to
+  force git to use SSH instead of https when fetching from GitHub.
+
+```sh
+[url "ssh://git@github.com/"]
+  insteadOf = https://github.com/
+```
+
+* `--offline`: The new tool doesn't maintain a git cache so we don't have the
+  option to use locally cached repositories.
+* `--no-requirements`
+* `--insecure`
+
+</span><br>
+
+### Configuring an Mbed program
+
+<span style="background-color:#ccffcc;">The subcommand `mbed-tools configure`
+generates an Mbed OS config CMake file named `mbed_config.cmake` in the path
+`cmake_build/<TARGET_NAME>/\<PROFILE>/\<TOOLCHAIN>/`.</span><br>
+
+### Compiling an Mbed program
+
+<span style="background-color:#ffcccc;">The subcommand `mbed-tools
+compile` can be used to compile the program. The command-line-option `mbed
+compile --source` is deprecated. To control which libraries and directories you
+want to build either modify your application's `CMakeLists.txt` to add/exclude
+or use `cmake --build cmake_build --target <library_name> --target
+<other_library_name>` to add any number of targets to your build.</span><br>
+
+<span style="background-color:#ffcccc;">The command-line-option `mbed
+compile --library` is deprecated. Use standard CMake `add_library()`
+instead.</span><br>
+
+<span style="background-color:#ffcccc;">The command-line-option `mbed compile
+-DMACRO_NAME` is deprecated. Use `mbed-tools configure -DMACRO_NAME` followed
```

Comment:
`mbed-tools configure` doesn't let you pass in macros, does it? You have to run a cmake configure directly to pass them in on the command line. `mbed-tools configure` generates `mbed_config.cmake` and the build directory, but it doesn't run the cmake configure step to generate the build system.

Reply:
I'll describe how to do this with CMake commands, then.